### PR TITLE
Fix #10729

### LINKED ISSUE #10729: incopatible type with typescript v2.9.1
- [ ] I have searched the [issues](https://github.com/ant-design/ant-design/issues) of this repository and believe that this is not a duplicate.

### Version
3.5.4

### Environment
bowser 3.5.4

### Reproduction link
[Edit on CodeSandbox](https://jnw46698m3.codesandbox.io/)

### Steps to reproduce
npm start

### What is expected?
compile successfully

### What is actually happening?
Error: 
antd\lib\_util\type.d.ts
(8,63): error TS2344: Type 'keyof T' does not satisfy the constraint 'string'.
  Type 'string | number | symbol' is not assignable to
type 'string'.
    Type 'number' is not assignable to type 'string'.

### PR DESCRIPTION
**Pull request by karol-majewski**
* [x] Make sure that you propose PR to right branch: bugfix for `master`, feature for latest active branch `feature-x.x`.
* [x] Make sure that you follow antd's [code convention](https://github.com/ant-design/ant-design/wiki/Code-convention-for-antd).
* [x] Run `npm run lint` and fix those errors before submitting in order to keep consistent code style.
* [x] Rebase before creating a PR to keep commit history clear.
* [x] Add some descriptions and refer relative issues for you PR.

### CODE REVIEW

File: components/_util/type.ts [hunk 1]
```diff
@@ -1,3 +1,3 @@
 export type Diff<T extends string, U extends string> = ({ [P in T]: P } &
```
Comment (picodoth) [inline components/_util/type.ts]: Diff not used anywhere now.

**Review by yesmeck — CHANGES_REQUESTED** [review]

File: components/_util/type.ts (hunk 1 above)
Comment (karol-majewski) [reply components/_util/type.ts]: It's not used, but it's exported. Are we sure removing it won't break anyone's build?

Comment (yesmeck) [reply components/_util/type.ts]: It's not a public API, we can remove it.

Comment (karol-majewski) [reply components/_util/type.ts]: Coming right up 🚀

**Comment by yesmeck** [comment]
Thanks!

**Comment by afc163** [comment]
close #10729

**Pull request merged**